Framework/LiveData/src/Kafka/KafkaHistoListener.cpp

```diff
@@ -0,0 +1,91 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2008 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#include "MantidLiveData/Kafka/KafkaHistoListener.h"
+
+#include "MantidAPI/LiveListenerFactory.h"
+#include "MantidLiveData/Exception.h"
+#include "MantidLiveData/Kafka/KafkaBroker.h"
+#include "MantidLiveData/Kafka/KafkaHistoStreamDecoder.h"
+#include "MantidLiveData/Kafka/KafkaTopicSubscriber.h"
+
+namespace {
+Mantid::Kernel::Logger g_log("KafkaHistoListener");
+}
+
+namespace Mantid {
+namespace LiveData {
+
+DECLARE_LISTENER(KafkaHistoListener)
+
+KafkaHistoListener::KafkaHistoListener() {
+  declareProperty("InstrumentName", "");
+}
+
+/// @copydoc ILiveListener::connect
+bool KafkaHistoListener::connect(const Poco::Net::SocketAddress &address) {
+  auto broker = std::make_shared<KafkaBroker>(address.toString());
```

Comment:
Why is the broker shared? Is this required?

Reply:
Copy-pasted from KafkaEventListener. 

Looks like it was unique at first, but was changed to shared by @matthew-d-jones in https://github.com/mantidproject/mantid/commit/00714fbbc540fd2b2bd1b70b254a73876a82b19c as part of #18450.

Reply:
I think I may have had problems with gmock when doing dependency injection with `unique_ptr`.

Reply:
Where exactly @matthew-d-jones?

Reply:
Would be worth a retry. I assume you are ending up (in the mock) pointing to invalid memory after the `unique_ptr` is moved. Perhaps avoiding `make_unique` or at least keeping the address of the held pointer is a better way to construct the mock?